core/src/main/java/de/evoila/cf/broker/util/ServiceInstanceUtils.java

```diff
@@ -122,10 +119,19 @@ public static boolean wouldCreateIdenticalInstance(String serviceInstanceId, Ser
                 && request.getPlanId().equals(serviceInstance.getPlanId())
                 && request.getOrganizationGuid().equals(serviceInstance.getOrganizationGuid())
                 && request.getSpaceGuid().equals(serviceInstance.getSpaceGuid())
-                && request.getContext().equals(serviceInstance.getContext())
+                && compareContext(request, serviceInstance)
                 && request.getParameters().equals(serviceInstance.getParameters());
     }
 
+    /*
+     * calls context.equals if the ServiceInstanceRequest contains a context Object and returns the. If the context object
+     * from the request the method returns true if the context object from serviceInstance is also null.
+     *
+     */
+    private static boolean compareContext(ServiceInstanceRequest request, ServiceInstance serviceInstance) {
+        return Optional.ofNullable(request.getContext()).map(context -> context.equals(serviceInstance.getContext()))
```

Comment:
This again is way too complicated and bloated for its actual value. Please keep the code readable!

Reply:
Replaced with classic null-checks